README.md

```diff
@@ -72,6 +72,18 @@ repository:
   # vulnerability alerts.
   enable_vulnerability_alerts: true
 
+  # Either `true` to make this repo available as a template repository or `false` to prevent it.
+  # is_template: false
+  # template_repository: false
+
+  # Note: You cannot unarchive repositories through the API. `true` to archive this repository. 
+  archived: false
+
+  # Can be `public`, `private` or `internal`
+  visibility: public
```

Comment:
does this conflict with the existing `private` attribute? is the use of `private` vs `visibility` mutually exclusive?

Reply:
they are different and it depends whether you are github enterprise user and what is the organization (within the enterprise) policy.

e.g. the organization set the default repository visibility to private, even if one will use `private: false` it won't affect the organization policy. yet, you can set `visibility` to `internal` or `private`.

here are some references for clarity (i don't want to copy and paste):
- https://developer.github.com/changes/2019-12-03-internal-visibility-changes/
- https://docs.github.com/en/rest/reference/repos#create-an-organization-repository--parameters

Reply:
i guess what i'm getting at is that i think we should clarify what happens when both `private` and `visibility` are defined together and maybe discourage defining both. from https://docs.github.com/en/rest/reference/repos#update-a-repository: 
> The visibility parameter overrides the private parameter when you use both along with the nebula-preview preview header.

> and it depends whether you are github enterprise user and what is the organization (within the enterprise) policy.

i think this is the other piece that should be mentioned here since `internal` doesnt apply in all cases. again from https://docs.github.com/en/rest/reference/repos#update-a-repository, i like the way that is described:
> Can be public or private. If your organization is associated with an enterprise account using GitHub Enterprise Cloud or GitHub Enterprise Server 2.20+, visibility can also be internal.